acestep/training_v2/ui/presets.py

```
@@ -0,0 +1,347 @@
+
+
+def _sanitize_name(name: str) -> str:
+    """Sanitize a preset name to be a safe filename stem.
+
+    Strips whitespace, replaces spaces with underscores, removes unsafe
+    characters, rejects path traversal attempts, and blocks Windows
+    reserved filenames (CON, NUL, COM1, etc.).
+    """
+    name = name.strip()
+    if not name:
+        raise ValueError("Preset name cannot be empty")
```

Ruff (0.15.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)